webrender/src/device/gfx/device.rs

```diff
@@ -1403,42 +1449,49 @@ impl<B: hal::Backend> Device<B> {
         );
     }
 
-    pub fn bind_per_draw_textures(&mut self) {
+    pub fn bind_per_draw_textures(&mut self, can_skip_bind: bool) {
         debug_assert!(self.inside_frame);
         assert_ne!(self.bound_program, INVALID_PROGRAM_ID);
+        let descriptor_group = self.bound_program.1;
+
+        if !can_skip_bind {
+            let per_draw_bindings = PerDrawBindings(
+                [
+                    self.bound_textures[0],
+                    self.bound_textures[1],
+                    self.bound_textures[2],
+                ],
+                [
+                    self.bound_sampler[0],
+                    self.bound_sampler[1],
+                    self.bound_sampler[2],
+                ],
+            );
 
-        let descriptor_group = {
-            let program = self
-                .programs
-                .get(&self.bound_program)
-                .expect("Program not found.");
-            program.shader_kind.into()
-        };
-        let per_draw_bindings = PerDrawBindings(
-            [
-                self.bound_textures[0],
-                self.bound_textures[1],
-                self.bound_textures[2],
-            ],
-            [
-                self.bound_sampler[0],
-                self.bound_sampler[1],
-                self.bound_sampler[2],
-            ],
-        );
+            if self.bound_per_draw_textures != per_draw_bindings {
+                let descriptor = self.bind_per_draw_textures_impl(descriptor_group, per_draw_bindings);
 
-        if self.bound_per_draw_textures == per_draw_bindings {
-            return;
+                Self::push_back(
```

Comment:
we need to rename this method to something more descriptive

Reply:
Renamed these in https://github.com/szeged/webrender/pull/350/commits/b0be810ff981417d41a2600b6a4cd74ae5f9a2af